experimental/collator/src/lib.rs

```diff
@@ -0,0 +1,1793 @@
+        }
+    }
+
+    #[test]
+    fn test_basic() {
+        let data_provider = icu_testdata::get_provider();
+
+        let mut options = CollatorOptions::new();
+        options.set_strength(Some(Strength::Quaternary));
+
+        let collator: Collator =
+            Collator::try_new(Locale::default(), &data_provider, options).unwrap();
+        assert_eq!(collator.compare("ac", "äb"), Ordering::Greater);
+    }
+
+    #[test]
+    fn test_implicit_unihan() {
+        // Adapted from `CollationTest::TestImplicits()` in collationtest.cpp of ICU4C.
+        // The radical-stroke order of the characters tested agrees with their code point
+        // order.
+        let data_provider = icu_testdata::get_provider();
+
+        let mut options = CollatorOptions::new();
+        options.set_strength(Some(Strength::Quaternary));
+
+        let collator: Collator =
+            Collator::try_new(Locale::default(), &data_provider, options).unwrap();
+        assert_eq!(collator.compare("\u{4E00}", "\u{4E00}"), Ordering::Equal);
+        assert_eq!(collator.compare("\u{4E00}", "\u{4E01}"), Ordering::Less);
+        assert_eq!(collator.compare("\u{4E01}", "\u{4E00}"), Ordering::Greater);
+
+        assert_eq!(collator.compare("\u{4E18}", "\u{4E42}"), Ordering::Less);
+        assert_eq!(collator.compare("\u{4E94}", "\u{50F6}"), Ordering::Less);
+    }
+
+    #[test]
+    fn test_currency() {
+        // Adapted from `CollationCurrencyTest::currencyTest` in currcoll.cpp of ICU4C.
+        // All the currency symbols, in collation order.
+        let currencies = "\u{00A4}\u{00A2}\u{FFE0}\u{0024}\u{FF04}\u{FE69}\u{00A3}\u{FFE1}\u{00A5}\u{FFE5}\u{09F2}\u{09F3}\u{0E3F}\u{17DB}\u{20A0}\u{20A1}\u{20A2}\u{20A3}\u{20A4}\u{20A5}\u{20A6}\u{20A7}\u{20A9}\u{FFE6}\u{20AA}\u{20AB}\u{20AC}\u{20AD}\u{20AE}\u{20AF}";
+        let data_provider = icu_testdata::get_provider();
+
+        let mut options = CollatorOptions::new();
+        options.set_strength(Some(Strength::Quaternary));
+
+        let collator: Collator =
+            Collator::try_new(Locale::default(), &data_provider, options).unwrap();
+        // Iterating as chars and re-encoding due to
+        // https://github.com/rust-lang/rust/issues/83871 being nightly-only. :-(
+        let mut lower_buf = [0u8; 4];
+        let mut higher_buf = [0u8; 4];
+        let mut chars = currencies.chars();
+        while let Some(lower) = chars.next() {
+            let tail = chars.clone();
+            for higher in tail {
+                let lower_str = lower.encode_utf8(&mut lower_buf);
+                let higher_str = higher.encode_utf8(&mut higher_buf);
+                assert_eq!(collator.compare(lower_str, higher_str), Ordering::Less);
+            }
+        }
+    }
+
+    #[test]
+    fn test_de() {
+        // Adapted from `CollationGermanTest` in decoll.cpp of ICU4C.
+        let left = [
+            "\u{47}\u{72}\u{00F6}\u{00DF}\u{65}",
+            "\u{61}\u{62}\u{63}",
```

Comment:
Markus: please unescape easily readable characters because it obscures what is happening.  Only use escapes when it helps readability (combining marks, invisible characters, the Han characters above in your sort-by-code-point test...).

This would be `"abc"`

Reply:
Filed as #1906